### PR TITLE
feat: Integrate Workflow Manger V2

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary

This PR brings the integration of Workflow Manger package defined in the https://github.com/OpenNSW/go-temporal-workflow. This package introduces the Node and Edges based graph representation for the workflow management with token passing through the edges to unlock new workflow nodes. And it make use of [temporal.io](https://temporal.io) for managing states of the workflow nodes.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- Added configuration option to enable the Workflow Manager V2  in the application.
- Updated the TaskManager interface to support callbacks for task completion and updates using the v2 workflow manager.
- Implemented logic in the ConsignmentService to handle workflow initialization and retrieval for both v1 and v2 workflows.
- Introduced a new JSON template for the Customs workflow.
- Enhanced unit tests to accommodate the workflow manager v2 integration.
- Add docker compose to run the temporal locally

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

### Testing Guidelines
1. Run `cp .env.exampl .env` in the root
2.  Set the `USE_WORKFLOW_MANAGER_V2` as,
    - `false` to validate v1 workflow manager functioning properly without any breaking
    - 'true` to check the functionality of the v2 workflow manager
3. Run `./start-dev.sh`. This will brings up all the services to validate end-to-end flow

## Additional Notes

This only bring the new version of the workflow manager such that previous functionality also exists. You need to setup the boolean environment `USE_WORKFLOW_MANAGER_V2` as `true` to make use of new version of  Workflow Manger package

## Deployment Notes
- Need to deploy Self-hosted Temporal Service if we setup `USE_WORKFLOW_MANAGER_V2` as `true`. Guide available at https://docs.temporal.io/self-hosted-guide

## Limitations need to be address in future PR
- The current workflow management is only wired for Consignment Workflow. And it's still not stable.

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

No specific issues created for this PR

## Screenshots/Demo

N/A (backed change)

### CODE REVIEW

File: backend/internal/app/bootstrap/app.go [hunk 1]
```diff
@@ -49,6 +53,73 @@ func writeJSON(w http.ResponseWriter, status int, v any) {
...
+	activationHandler := func(payload workflowManagerV2.TaskPayload) error {
+		template, err := templateService.GetWorkflowNodeTemplateByID(ctx, payload.TaskTemplateID)
+		if err != nil {
+			return fmt.Errorf("Error getting workflow node template: %v\n", err)
+		}
+
+		// TODO: We need to pass the TaskPayload.RunID in the future to avoid issues with
+		// task retries. For example, when retrying a task instance, a stale version might
+		// send a completion that will trigger the new version.
+		slog.Error("---- DEBUG ---- Starting InitTaskRequest", "TaskID", payload.NodeID, "WorkflowID", payload.RunID, "WorkflowNodeTemplateID", template.ID)
```
[the remark this answers is not in the document]
Comment (lokewate) [reply backend/internal/app/bootstrap/app.go]: please remove these debug messages that I had accidentally checked in.
